### PR TITLE
CASMPET-5898 Split OPA policies by use case

### PR DESCRIPTION
**Pull request by kburns-hpe**
## Summary and Scope

This changes the OPA policies to be split by use case instead of gateway. It also allows custom OPA policies to be created without modifying the existing policies. This allows customers to add their own policies without worrying about them being wiped out by an upgrade.

The policies are backwards compatible with the previous cray-opa version.

## Issues and Related PRs

* Resolves [CASMPET-5898](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-5898)

### Tested on:

  * Local development environment
  * Virtual Shasta

### Test description:

Tests were done via unit tests, kuttl tests, and in vshasta. Policies were validated to be working after an install.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? Y
- Were continuous integration tests run? If not, why? Y
- Was upgrade tested? If not, why? Y
- Was downgrade tested? If not, why? Y
- Were new tests (or test issues/Jiras) created for this change? Y

## Risks and Mitigations

This needs further testing on bare metal.

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [ ] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by kburns-hpe** [comment]
Jenkins is timing out trying to pull artifacts from algol60. I'll kick off a new test in a bit and see if it clears up.

File: kubernetes/cray-opa/README.md [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+this, set `.spec.ingresses.[INGRESS GATEWAY].custom` to a list containing the
+ConfigMaps that hold the policy modules you wish to apply. Each module needs
+to have the package name `istio.authz`. The file name in the ConfigMap should be
+named `policy.rego`.
 
-Note: Make sure the image in the Dockerfile matches the actual OPA image used, check the version.
+## Testing
+
+Tests are run using `make test`.
+
```
Comment (jeremy-duckworth) [inline kubernetes/cray-opa/README.md:17]: Might help to list the policy 'files' out and provide some context around the top-level values settings, guidance to other teams re: what policy they might need to update, etc.

**Review by rnoska-hpe — APPROVED** [review]
Approved pending resolution of the rego test issue.

**Review by jeremy-duckworth — APPROVED** [review]
Very cool to see this. 

Chatted via DM about https://github.com/open-policy-agent/kube-mgmt and possible future iterations. Minor feedback.

**Review by rgross727 — APPROVED** [review]

**Pull request merged**